clippy_lints/src/methods/iter_kv_map.rs

```diff
@@ -21,7 +22,11 @@ pub(super) fn check<'tcx>(
     expr: &'tcx Expr<'tcx>,  // .iter().map(|(_, v_| v))
     recv: &'tcx Expr<'tcx>,  // hashmap
     m_arg: &'tcx Expr<'tcx>, // |(_, v)| v
+    msrv: &Msrv,
 ) {
+    if map_type == "into_iter" && !msrv.meets(msrvs::INTO_KEYS) {
+        return;
+    }
     if !expr.span.from_expansion()
         && let ExprKind::Closure(c) = m_arg.kind
```

Comment:
This lint should probably check that the closure doesn't have parameter- or return type annotations since that can help type inference or make the `map` closure act as a coercion site.
Not your problem though, that could be a PR on its own, unless you also want to try to fix that here ^^.

Examples where the machine applicable suggestion breaks: [coercions](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d7178f8b562958d52c82af1e40091ff0), [inference](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a67abf071b2ef645d886e8cab5fa2da1)

Reply:
I will have a try fixing this, but  I think I will open a separate PR so this one doesn't get too cluttered ^^